[dnm] ci: write artifacts for individual failing tests

tbg opened the pull request:
Just grabbing a ref at this point to run my experiments. Nothing to see here.

tbg committed:
delme

tbg committed:
build: write artifacts/failures.txt in CI

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/42538)
<!-- Reviewable:end -->

tbg committed:
acceptance: don't swallow stdout

<what was there before: Previously, ...>
<why it needed to change: This was inadequate because ...>
<what you did about it: To address this, this patch ...>

Release note (<category, see below>): <what> <show> <why>
